### PR TITLE
fix(proto): avoid sharing state

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/grpc-transcode/proto.lua [hunk 1]
```diff
@@ -29,6 +30,10 @@ local lrucache_proto = core.lrucache.new({
 local proto_fake_file = "filename for loaded"
 
 local function compile_proto(content)
+    -- clear pb state
+    pb.state(nil)
```
Comment (bisakhmondal) [inline apisix/plugins/grpc-transcode/proto.lua:34]: Isn't it redundant? I think Line 54 is already retrieving the current pb state and setting it to nil.

**Review by bisakhmondal — APPROVED** [review]
Great fix indeed :+1:

**Review by tzssangglass — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

File: apisix/plugins/grpc-transcode/proto.lua (hunk 1 above)
Comment (spacewander) [reply apisix/plugins/grpc-transcode/proto.lua:34]: There is a default pb state that we need to clear.

Comment (bisakhmondal) [reply apisix/plugins/grpc-transcode/proto.lua:34]: Got it.

**Pull request merged**